src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -159,12 +162,27 @@ public void onAtomicStep(@Nonnull StepAtomNode node, @Nonnull WorkflowRun run) {
                 spanBuilder = getTracer().spanBuilder(node.getDisplayFunctionName());
             }
 
+            String pluginName = "unknown";
+            String pluginVersion = "unknown";
+            try {
+                Jenkins jenkins = Jenkins.getInstanceOrNull();
+                if (jenkins != null) {
+                    PluginWrapper wrapper = jenkins.getPluginManager().whichPlugin(Objects.requireNonNull(((StepAtomNode) node).getDescriptor()).clazz);
+                    pluginName = wrapper.getShortName();
+                    pluginVersion = wrapper.getVersion();
+                }
```

Comment:
This section could be somehow moved to a Global cache map to avoid the cost of calling the `whichPlugin`. What do you think?